modules/account-lib/src/coin/trx/wrappedBuilder.ts

```diff
@@ -0,0 +1,92 @@
+import { Transaction } from './transaction';
+import { BaseKey } from '../baseCoin/iface';
+import { Address } from './address';
+import BigNumber from 'bignumber.js';
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { TransactionBuilder } from './transactionBuilder';
+import { BaseTransaction } from '../baseCoin';
+import { KeyPair } from './keyPair';
+import { decodeTransaction } from './utils';
+import { ContractType } from './enum';
+import { InvalidTransactionError } from '../baseCoin/errors';
+import { ContractCallBuilder } from './contractCallBuilder';
+
+export class WrappedBuilder extends TransactionBuilder {
+  private _builder: TransactionBuilder;
+  // private _builder: TransactionBuilder | ContractCallBuilder;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+    // defaults to old builder
+    this._builder = new TransactionBuilder(_coinConfig);
+  }
+
+  getContractCallBuilder(tx?: Transaction): ContractCallBuilder {
+    return this.initializeBuilder(tx, new ContractCallBuilder(this._coinConfig));
+  }
+
+  private initializeBuilder<T extends TransactionBuilder>(tx: Transaction | undefined, builder: T): T {
+    if (tx) {
+      builder.initBuilder(tx);
+    }
+    return builder;
+  }
+  extendValidTo(extensionMs: number) {
+    this._builder.extendValidTo(extensionMs);
+  }
+
+  sign(key: BaseKey) {
+    this._builder.sign(key);
+  }
+
+  async build(): Promise<BaseTransaction> {
+    return this._builder.build();
+  }
+
+  from(raw: any) {
+    this.validateRawTransaction(raw);
+    const rawDataHex = this.getTxReceip(raw);
+    const decodedTx = decodeTransaction(rawDataHex);
+    const contractType = decodedTx.contractType;
+    switch (contractType) {
+      case ContractType.Transfer:
+      case ContractType.AccountPermissionUpdate:
+        this._builder = new TransactionBuilder(this._coinConfig);
+        break;
+      case ContractType.TriggerSmartContract:
+        return this.getContractCallBuilder(raw);
```

Comment:
let's try to keep this pattern consistent. we are calling `this._builder.from()` below. this doesn't really makes sense as a normal switch

Reply:
thanks, I moved that return inside the case option

Reply:
Done